cocos/spine/skeleton.ts

```diff
@@ -735,828 +619,461 @@ export class Skeleton extends UIRenderer {
+        Enum.update(this._enumSkins);
+        setEnumAttr(this, '_defaultSkinIndex', this._enumSkins);
     }
 
-    /**
-     * @en Sets the end event listener for specified TrackEntry.
-     * @zh 用来为指定的 TrackEntry 设置动画播放结束的事件监听。
-     * @method setTrackEndListener
-     * @param {sp.spine.TrackEntry} entry
-     * @param {function} listener
-     */
-    public setTrackEndListener (entry: spine.TrackEntry, listener: TrackListener) {
-        TrackEntryListeners.getListeners(entry).end = listener;
+    protected _refreshInspector () {
+        if (EDITOR && !legacyCC.GAME_VIEW) {
+            // update inspector
+            this._updateAnimEnum();
+            this._updateSkinEnum();
+            // TODO: refresh inspector
+            // Editor.Utils.refreshSelectedInspector('node', this.node.uuid);
+        }
     }
 
     /**
-     * @en Sets the dispose event listener for specified TrackEntry.
-     * @zh 用来为指定的 TrackEntry 设置动画即将被销毁的事件监听。
-     * @method setTrackDisposeListener
-     * @param {sp.spine.TrackEntry} entry
-     * @param {function} listener
+     * @en Call this method to destroy the rendering data.
+     * @zh 调用该方法销毁渲染数据。
      */
-    public setTrackDisposeListener (entry: spine.TrackEntry, listener: TrackListener) {
-        TrackEntryListeners.getListeners(entry).dispose = listener;
+    public destroyRenderData () {
+        this._drawList.reset();
+        super.destroyRenderData();
     }
 
-    /**
-     * @en Sets the complete event listener for specified TrackEntry.
-     * @zh 用来为指定的 TrackEntry 设置动画一次循环播放结束的事件监听。
-     * @method setTrackCompleteListener
-     * @param {sp.spine.TrackEntry} entry
-     * @param {function} listener
-     * @param {sp.spine.TrackEntry} listener.entry
-     * @param {Number} listener.loopCount
-     */
-    public setTrackCompleteListener (entry: spine.TrackEntry, listener: TrackListener2) {
-        TrackEntryListeners.getListeners(entry).complete = function (trackEntry) {
-            const loopCount = Math.floor(trackEntry.trackTime / trackEntry.animationEnd);
-            listener(trackEntry, loopCount);
-        };
+    protected createRenderEntity () {
+        const renderEntity = new RenderEntity(RenderEntityType.DYNAMIC);
+        renderEntity.setUseLocal(true);
+        return renderEntity;
     }
-
     /**
-     * @en Sets the event listener for specified TrackEntry.
-     * @zh 用来为指定的 TrackEntry 设置动画帧事件的监听。
-     * @method setTrackEventListener
-     * @param {sp.spine.TrackEntry} entry
-     * @param {function} listener
+     * @en Mark to re-update the rendering data, usually used to force refresh the display.
+     * @zh 标记重新更新渲染数据，一般用于强制刷新显示。
      */
-    public setTrackEventListener (entry: spine.TrackEntry, listener: TrackListener|TrackListener2) {
-        TrackEntryListeners.getListeners(entry).event = listener;
+    public markForUpdateRenderData (enable = true) {
+        super.markForUpdateRenderData(enable);
```

Comment:
If no other override need, we should remove this function and use the inherited function directly

Reply:
So the debug renderer won't be updated currently ?